lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -47,7 +46,7 @@ def __init__(
 
         super().__init__(charm, self.relation_name)
         self.framework.observe(
-            charm.on[self.relation_name].relation_joined, self._on_relation_joined
+            charm.on[self.relation_name].relation_changed, self._on_relation_changed
```

Comment:
this way if the `mongos` user changes its requested roles/databases the function is called and updates the user